WcaOnRails/app/controllers/admin_controller.rb

```diff
@@ -75,4 +75,16 @@ def person_data
       dob: @person.dob,
     }
   end
+
+  def compute_auxiliary_data
+    @auxiliary_data_start = Timestamp.find_or_create_by(name: 'auxiliary_data_start')
+    @auxiliary_data_end = Timestamp.find_or_create_by(name: 'auxiliary_data_end')
+  end
+
+  def do_compute_auxiliary_data
+    # Reset the end date. The start date is updated directly before the computation.
+    Timestamp.find_or_create_by(name: 'auxiliary_data_end').update date: nil
+    ComputeAuxiliaryData.perform_later
```

Comment:
A bit tricky that raising an exception doesn't cause the job to fail. That said I love that generalization and will give it a try =)